IdentityCore/src/telemetry/request_telemetry/MSIDLastRequestTelemetrySerializedItem.m

```diff
@@ -0,0 +1,110 @@
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#import "MSIDCurrentRequestTelemetrySerializedItem+Internal.h"
+#import "MSIDLastRequestTelemetrySerializedItem.h"
+
+@interface MSIDLastRequestTelemetrySerializedItem()
+
+@property (nonatomic) NSArray<MSIDRequestTelemetryErrorInfo *> *errorsInfo;
+
+@end
+
+@implementation MSIDLastRequestTelemetrySerializedItem
+
+- (instancetype)initWithSchemaVersion:(NSNumber *)schemaVersion defaultFields:(NSArray *)defaultFields errorInfo:(NSArray *)errorsInfo platformFields:(NSArray *)platformFields
+{
+    self = [super initWithSchemaVersion:schemaVersion defaultFields:defaultFields platformFields:platformFields];
+    if (self)
+    {
+        _errorsInfo = errorsInfo;
+    }
+    return self;
+}
+
+// Builds last telemetry string using default serialization of each set of fields
+// specified in the last telemetry string schema
+- (NSString *)serialize
+{
+    NSString *telemetryString = [NSString stringWithFormat:@"%@|", self.schemaVersion];
+    telemetryString = [telemetryString stringByAppendingFormat:@"%@|", [super serializeFields: super.defaultFields]];
+    
+    NSUInteger startLength = [telemetryString lengthOfBytesUsingEncoding:NSUTF8StringEncoding];
+    telemetryString = [telemetryString stringByAppendingFormat:@"%@|", [self serializeErrorsInfoWithCurrentStringSize:startLength]];
+    
+    telemetryString = [telemetryString stringByAppendingFormat:@"%@", [super serializeFields: self.platformFields]];
+    
+    return telemetryString;
+}
+
+#pragma mark Helper
+
+- (NSString *)serializeErrorsInfoWithCurrentStringSize:(NSUInteger)startLength
+{
+    NSString *failedRequestsString = @"";
+    NSString *errorMessagesString = @"";
+    
+    if (self.errorsInfo.count > 0)
+    {
+        int lastIndex = (int)self.errorsInfo.count - 1;
+        
+        // Set first post in fencepost structure -- last item in string doesn't have comma at the end
+        NSString *currentFailedRequest = [NSString stringWithFormat:@"%ld,%@", self.errorsInfo[lastIndex].apiId, self.errorsInfo[lastIndex].correlationId ?: @""];
+        NSString *currentErrorMessage = [NSString stringWithFormat:@"%@", self.errorsInfo[lastIndex].error ?: @""];
+        
+        // Only add error info into string if the resulting string smaller than 4KB
+        if ([currentFailedRequest lengthOfBytesUsingEncoding:NSUTF8StringEncoding] +
+                [currentErrorMessage lengthOfBytesUsingEncoding:NSUTF8StringEncoding] + startLength < 4000)
+        {
+            failedRequestsString = [currentFailedRequest stringByAppendingString:failedRequestsString];
+            errorMessagesString = [currentErrorMessage stringByAppendingString:errorMessagesString];
+        }
+
+        // Fill in remaining errors with comma at the end of each error
+        for (int i = lastIndex - 1; i >= 0; i--)
```

Comment:
would be great to see unit tests for a case with more than one error info

Reply:
That's a great point. I added a few more tests to cover cases with multiple errors in errorInfo.